libs/MeshKernel/include/MeshKernel/TriangulationWrapper.hpp

```diff
@@ -83,117 +73,165 @@ namespace meshkernel
+        /// @param edgeIndex The index of the edge to retrieve
+        /// @return const reference to the edge with the specified index for the specified face
+        [[nodiscard]] const auto& GetFaceEdge(const size_t& faceIndex, const size_t& edgeIndex) const
+        {
+            return m_faceEdges[faceIndex][edgeIndex];
+        }
+
+        /// @brief Retrieves the edge node
+        /// @param edgeIndex The index of the edge to retrieve the node from
+        /// @param nodeIndex The index of the node to retrieve
+        /// @return const reference to the node with the specified index for the specified face
+        [[nodiscard]] const auto& GetEdgeNode(const size_t& edgeIndex, const size_t& nodeIndex) const
+        {
+            return m_edgeNodes[edgeIndex][nodeIndex];
+        }
+
+        /// @brief Retrieves the edge face
+        /// @param edgeIndex The index of the edge to retrieve the node from
+        /// @param faceIndex The index of the face to retrieve
+        /// @return const reference to the edge with the specified index for the specified face
+        [[nodiscard]] const auto& GetEdgeFace(const size_t& edgeIndex, const size_t& faceIndex) const
+        {
+            return m_edgesFaces[edgeIndex][faceIndex];
+        }
+
+        /// @brief Retrieves the x coordinate of a triangulated node
+        /// @param nodeIndex The index of the node to retrieve
+        /// @return const reference to the x coordinate
+        [[nodiscard]] const auto& GetXCoord(const size_t& nodeIndex) const
+        {
+            return m_xCoordFlat[nodeIndex];
+        }
+
+        /// @brief Retrieves the y coordinate of a triangulated node
+        /// @param nodeIndex The index of the node to retrieve
+        /// @return const reference to the y coordinate
+        [[nodiscard]] const auto& GetYCoord(const size_t& nodeIndex) const
+        {
+            return m_yCoordFlat[nodeIndex];
         }
+
+    private:
+
+        std::vector<int> m_faceNodesFlat; ///< Face nodes flat array passed to the triangulation library
+        std::vector<int> m_edgeNodesFlat; ///< Edge nodes flat array passed to the triangulation library
+        std::vector<int> m_faceEdgesFlat; ///< Face edges flat array passed to the triangulation library
+        std::vector<double> m_xCoordFlat; ///< x coordinates flat array passed to the triangulation library
+        std::vector<double> m_yCoordFlat; ///< y coordinates flat array passed to the triangulation library
+        int m_numNodes{0};                ///< Initial number of triangulated nodes
+        int m_numEdges{0};                ///< Initial number of triangulated edges
+        int m_numFaces{0};                ///< Initial number of triangulated faces
+
+        std::vector<Point> m_nodes;                    ///< Reconstructed vector of nodes
+        std::vector<std::vector<size_t>> m_faceNodes;  ///< Reconstructed vector of face nodes
```

Comment:
Nothing to to do, just a note.
I've seen vector of vectors used a lot throughout the code. We can:
1. define out own matrix class Matrix (contiguous memory) with any necessary operators.
2. use boost::numeric::ublas::matrix
3. use the eigen library

Reply:
indeed, i agree